### PR TITLE
Reducer script for non-ctu findings.

### PR DESCRIPTION
**Pull request by bruntib**
This script attempts to reduce a source code which still produces the
given CodeChecker finding. The documentation can be found on the --help
page of the script.

### CODE REVIEW

File: scripts/debug_tools/reduce.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+import tempfile
+
+
+def check_files_existence(report_dir):
+    '''
+    This function checks the existence of some files which are required for the
+    reduction process.
+    '''
+    def check(path, isfile):
+        if not (os.path.isfile(path) if isfile else os.path.isdir(path)):
```
Comment (csordasmarton) [inline scripts/debug_tools/reduce.py]: Couldn't we use `os.path.exists` here: https://docs.python.org/2/library/os.path.html

File: scripts/debug_tools/reduce.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        analysis.
+    workspace_dir -- The workspace directory which contains the new report
+        directory which is used for reduction. The new report directory should
+        be a temporary location because its content is rewritten between the
+        reduction steps.
+    codechecker -- Full path of CodeChecker.
+    specific_checker -- For fastening reduction only this checker will be
+        enabled and all others will be disabled
+    '''
+    def remove_flag(cmd, flag):
```
Comment (csordasmarton) [inline scripts/debug_tools/reduce.py]: We remove the parameter of the flag too. Maybe `remove_parameterized_flag` function name would be much better or create and extra argument for this function that allows the user to remove only the flag.

File: scripts/debug_tools/reduce.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+import re
+import shutil
+import stat
+import subprocess
+import sys
+import tempfile
+
+
+def check_files_existence(report_dir):
+    '''
```
Comment (csordasmarton) [inline scripts/debug_tools/reduce.py]: **Triple double-quoted** string should be used for docstring. My editor (pycharm) gives me this warning. Please replace this everywhere in this file.

File: scripts/debug_tools/reduce.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+# --- Command line arguments --- #
+
+parser = argparse.ArgumentParser(description='''
+Reduce for a specific bug.
+
+Given the following arguments this script attempts to create a minimal example
+which still produces a specific CodeChecker finding. Reduction steps are
+accomplished as long as the bug is still observable based on the output of
+CodeChecker parse command.''')
```
Comment (csordasmarton) [inline scripts/debug_tools/reduce.py]: Could you please write some usage examples in the epilog:
```py
parser = argparse.ArgumentParser(description='''...''',
                                 formatter_class=argparse.RawDescriptionHelpFormatter,
                                 epilog='''
Example scenario: Reduce source code
------------------------------------------------
TODO: write some usage example
''')
```
For example we use it at cmd:
https://github.com/Ericsson/codechecker/blob/7a28027bf19ad2348eba25455522f3021cd017ff/libcodechecker/libhandlers/cmd.py#L933

**Comment by dkrupp** [comment]
If we reduce only to a text in the plist, we do not need a script.

**Pull request closed without merging**